source/server/hot_restarting_child.h

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include "server/hot_restarting_base.h"
+
+namespace Envoy {
+namespace Server {
+
+/**
+ * The child half of hot restarting. Issues requests and commands to the parent.
+ */
+class HotRestartingChild : HotRestartingBase, Logger::Loggable<Logger::Id::main> {
+public:
+  HotRestartingChild(int base_id, int restart_epoch);
+
+  int duplicateParentListenSocket(const std::string& address);
+  std::unique_ptr<envoy::api::v2::core::HotRestartMessage> getParentStats();
+  void drainParentListeners();
+  void shutdownParentAdmin(HotRestart::ShutdownParentAdminInfo& info);
+  void terminateParent();
+  // Merge the values of stats_proto into stats_store. Counters are always straightforward
+  // addition, while gauges default to addition but have exceptions.
+  // Exceptions are listed in a map at the top of this function's body.
+  void mergeParentStats(Stats::Store& stats_store,
+                        const envoy::api::v2::core::HotRestartMessage::Reply::Stats& stats_proto);
+
+private:
+  const int restart_epoch_;
+  bool parent_terminated_{};
+  sockaddr_un parent_address_;
+
+  enum class CombineLogic {
```

Comment:
Please add doc comments for each of these. Also, per my other comments, I would like to lift all of the combination/filtering logic out of the hot restart files into a dedicated set of files.

Reply:
Done.